### PR TITLE
Appraiser-widget pagination

### PR DESCRIPTION
**Pull request by Space-tourist**

### CODE REVIEW

File: src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs [hunk 1]
```diff
@@ -18,9 +18,17 @@ public async Task<GetAssessmentHistoryResult> Handle(GetAssessmentHistoryQuery q
     {
         ArgumentNullException.ThrowIfNull(query);
 
+        DateTimeOffset? from = null;
+        
         var now = DateTimeOffset.UtcNow;
         var before = new DateTimeOffset(new DateOnly(now.Year, now.Month, now.Day), TimeOnly.MinValue, TimeSpan.Zero);
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs:24]: Rename variable 'before' -> 'to'

File: src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs [hunk 2]
```diff
@@ -18,9 +18,17 @@ public async Task<GetAssessmentHistoryResult> Handle(GetAssessmentHistoryQuery q
     {
         ArgumentNullException.ThrowIfNull(query);
 
+        DateTimeOffset? from = null;
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs:21]: Please use ternary expression 'var from = query.From is null ? null : new DateTimeOffset(query.From.Value, TimeOnly.MinValue, TimeSpan.Zero)'

File: src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs [hunk 3]
```diff
@@ -28,15 +28,14 @@ public StoryReader(IConnectionFactory connectionFactory)
                 COUNT(*) AS storiescount
             FROM appraiser.stories AS s
             WHERE s.team_id = @team_id AND s.created <= @before
+            AND ((@from is not null AND s.created >= @from) OR @from is null)
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs:31]: Please write simple '((@from is not null AND s.created >= @from) OR @from is null)' -> '(@from is null OR s.created >= @from)'

File: src/Inc.TeamAssistant.Gateway/wwwroot/langs/en.json [hunk 4]
```diff
@@ -170,6 +170,10 @@
   "Dashboard_TwoWeeks": "2 weeks",
   "Dashboard_FourWeeks": "4 weeks",
   "Dashboard_TwelveWeeks": "12 weeks",
+  "Dashboard_Month": "Month",
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Gateway/wwwroot/langs/en.json:173]: Use number of month: 1 month, 3 month, 6 month, 1 year

File: src/Inc.TeamAssistant.Reviewer.Application/CommandHandlers/MoveToReview/Validators/MoveToReviewCommandValidator.cs [hunk 5]
```diff
@@ -17,7 +17,15 @@ public MoveToReviewCommandValidator(ReviewerOptions reviewerOptions)
...
+            .Must(e => IsDescriptionConsistent(e, reviewerOptions))
+            .WithMessage("'{PropertyName}' must contains a link to the source code and some description");
+    }
+
+    private static bool IsDescriptionConsistent(string description, ReviewerOptions reviewerOptions)
+    {
+        string[] splittedDescription = description.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+        var links = splittedDescription.Where(t => reviewerOptions.LinksPrefix.Any(t.Contains)).ToArray();
+
+        return links.Any() && splittedDescription.Length > links.Length;
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Reviewer.Application/CommandHandlers/MoveToReview/Validators/MoveToReviewCommandValidator.cs:29]: Elegant solution.

File: src/Inc.TeamAssistant.Reviewer.Application/CommandHandlers/MoveToReview/Validators/MoveToReviewCommandValidator.cs [hunk 6]
```diff
@@ -17,7 +17,15 @@ public MoveToReviewCommandValidator(ReviewerOptions reviewerOptions)
             .MaximumLength(2000)
             .Must(e => !e.StartsWith("/"))
             .WithMessage("'{PropertyName}' please enter text value")
-            .Must(e => reviewerOptions.LinksPrefix.Any(e.Contains))
-            .WithMessage("'{PropertyName}' must contains a link to the source code");
+            .Must(e => IsDescriptionConsistent(e, reviewerOptions))
+            .WithMessage("'{PropertyName}' must contains a link to the source code and some description");
+    }
+
+    private static bool IsDescriptionConsistent(string description, ReviewerOptions reviewerOptions)
```
Comment (dyatlov-a) [inline src/Inc.TeamAssistant.Reviewer.Application/CommandHandlers/MoveToReview/Validators/MoveToReviewCommandValidator.cs:24]: Use more descriptive names. 'IsDescriptionConsistent' -> 'HasDescriptionAndLinks'

File: src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs (hunk 1 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs (hunk 2 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.Application/QueryHandlers/GetAssessmentHistory/GetAssessmentHistoryQueryHandler.cs]: fixed

File: src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs (hunk 3 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Appraiser.DataAccess/StoryReader.cs]: fixed

File: src/Inc.TeamAssistant.Gateway/wwwroot/langs/en.json (hunk 4 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Gateway/wwwroot/langs/en.json]: fixed

File: src/Inc.TeamAssistant.Reviewer.Application/CommandHandlers/MoveToReview/Validators/MoveToReviewCommandValidator.cs (hunk 6 above)
Comment (Space-tourist) [reply src/Inc.TeamAssistant.Reviewer.Application/CommandHandlers/MoveToReview/Validators/MoveToReviewCommandValidator.cs]: fixed

**Review by dyatlov-a — APPROVED** [review]

**Pull request merged**